pkg/eventingtls/eventingtlstesting/eventingtlstesting.go

```diff
@@ -42,8 +43,23 @@ func init() {
 	CA, Key, Crt = loadCerts()
 }
 
-func StartServer(ctx context.Context, t *testing.T, port int, requests chan<- *nethttp.Request) string {
+func StartServer(ctx context.Context, t *testing.T, port int, requests chan<- *nethttp.Request, receiverOptions ...kncloudevents.HTTPMessageReceiverOption) string {
+	handler := nethttp.HandlerFunc(func(writer nethttp.ResponseWriter, request *nethttp.Request) {
+		if requests != nil {
+			requests <- request
+		}
+		if request.TLS == nil {
+			// It's not on TLS, fail request
+			writer.WriteHeader(nethttp.StatusInternalServerError)
+			return
+		}
+		writer.WriteHeader(nethttp.StatusOK)
+	})
+
+	return StartServerWithCustomHandler(ctx, t, port, handler, receiverOptions...)
+}
 
+func StartServerWithCustomHandler(ctx context.Context, t *testing.T, port int, handler http.Handler, receiverOptions ...kncloudevents.HTTPMessageReceiverOption) string {
```

Comment:
I'd make a single function and always take an handler since `StartServer` seems less common than initially thought?

Reply:
we can always offer utilities that are `handler` for specific cases but the main function `StartServer` should be solving the common cases and the package doesn't become like other with many functions all with XWithY or XWithYAndZ

Reply:
:+1: Updated it